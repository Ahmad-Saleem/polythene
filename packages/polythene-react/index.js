export * from "polythene-react-base";
export * from "polythene-react-button";
export * from "polythene-react-card";
export * from "polythene-react-checkbox";
export * from "polythene-react-dialog";
export * from "polythene-react-dialog-pane";
export * from "polythene-react-fab";
export * from "polythene-react-icon";
export * from "polythene-react-icon-button";
export * from "polythene-react-list";
export * from "polythene-react-list-tile";
export * from "polythene-react-menu";
export * from "polythene-react-radio-button";
export * from "polythene-react-raised-button";
export * from "polythene-react-ripple";
export * from "polythene-react-shadow";
export * from "polythene-react-svg";
export * from "polythene-react-switch";
export * from "polythene-react-textfield";
export * from "polythene-react-toolbar";
