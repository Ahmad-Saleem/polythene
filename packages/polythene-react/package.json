{
  "name": "polythene-react",
  "version": "1.0.0-rc.8",
  "description": "",
  "main": "dist/polythene-react.js",
  "module": "dist/polythene-react.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build-standalone": "DEPS=1 ENTRY=scripts/standalone.js SOURCEMAP=1 DEST=./dist/polythene-react-standalone.js ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd-includes.js",
    "build": "yarn run rollup && yarn run build-standalone"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-css": "^1.0.0-rc.8",
    "polythene-react-base": "^1.0.0-rc.8",
    "polythene-react-button": "^1.0.0-rc.8",
    "polythene-react-card": "^1.0.0-rc.8",
    "polythene-react-checkbox": "^1.0.0-rc.8",
    "polythene-react-dialog": "^1.0.0-rc.8",
    "polythene-react-dialog-pane": "^1.0.0-rc.8",
    "polythene-react-fab": "^1.0.0-rc.8",
    "polythene-react-icon": "^1.0.0-rc.8",
    "polythene-react-icon-button": "^1.0.0-rc.8",
    "polythene-react-ios-spinner": "^1.0.0-rc.8",
    "polythene-react-list": "^1.0.0-rc.8",
    "polythene-react-list-tile": "^1.0.0-rc.8",
    "polythene-react-material-design-progress-spinner": "^1.0.0-rc.8",
    "polythene-react-material-design-spinner": "^1.0.0-rc.8",
    "polythene-react-menu": "^1.0.0-rc.8",
    "polythene-react-notification": "^1.0.0-rc.8",
    "polythene-react-radio-button": "^1.0.0-rc.8",
    "polythene-react-radio-group": "^1.0.0-rc.8",
    "polythene-react-raised-button": "^1.0.0-rc.8",
    "polythene-react-ripple": "^1.0.0-rc.8",
    "polythene-react-search": "^1.0.0-rc.8",
    "polythene-react-shadow": "^1.0.0-rc.8",
    "polythene-react-slider": "^1.0.0-rc.8",
    "polythene-react-snackbar": "^1.0.0-rc.8",
    "polythene-react-svg": "^1.0.0-rc.8",
    "polythene-react-switch": "^1.0.0-rc.8",
    "polythene-react-tabs": "^1.0.0-rc.8",
    "polythene-react-textfield": "^1.0.0-rc.8",
    "polythene-react-toolbar": "^1.0.0-rc.8"
  },
  "devDependencies": {
    "polythene-core": "^1.0.0-rc.8",
    "polythene-core-css": "^1.0.0-rc.8",
    "polythene-fastclick": "^1.0.0-rc.8",
    "react": "^15.6.1",
    "react-hyperscript": "^3.0.0"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://arthurclemens.com)",
  "license": "MIT"
}
