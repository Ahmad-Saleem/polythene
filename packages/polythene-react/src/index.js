export * from "./core";
export { button } from "./button";
export { icon } from "./icon";
export { ripple } from "./ripple";
export { shadow } from "./shadow";
export { svg } from "./svg";
