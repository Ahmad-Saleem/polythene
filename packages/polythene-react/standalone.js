export * from "./index";
import { addLayoutStyles } from "polythene-utilities";
import { addTypography, addRoboto } from "polythene-style";
import { addFastClick } from "polythene-fastclick";

addTypography();
addRoboto();
addLayoutStyles();
addFastClick();
