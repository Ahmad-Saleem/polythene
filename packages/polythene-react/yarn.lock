# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


asap@~2.0.3:
  version "2.0.5"
  resolved "https://registry.yarnpkg.com/asap/-/asap-2.0.5.tgz#522765b50c3510490e52d7dcfe085ef9ba96958f"

core-js@^1.0.0:
  version "1.2.7"
  resolved "https://registry.yarnpkg.com/core-js/-/core-js-1.2.7.tgz#652294c14651db28fa93bd2d5ff2983a4f08c636"

create-react-class@^15.6.0:
  version "15.6.0"
  resolved "https://registry.yarnpkg.com/create-react-class/-/create-react-class-15.6.0.tgz#ab448497c26566e1e29413e883207d57cfe7bed4"
  dependencies:
    fbjs "^0.8.9"
    loose-envify "^1.3.1"
    object-assign "^4.1.1"

encoding@^0.1.11:
  version "0.1.12"
  resolved "https://registry.yarnpkg.com/encoding/-/encoding-0.1.12.tgz#538b66f3ee62cd1ab51ec323829d1f9480c74beb"
  dependencies:
    iconv-lite "~0.4.13"

fbjs@^0.8.9:
  version "0.8.12"
  resolved "https://registry.yarnpkg.com/fbjs/-/fbjs-0.8.12.tgz#10b5d92f76d45575fd63a217d4ea02bea2f8ed04"
  dependencies:
    core-js "^1.0.0"
    isomorphic-fetch "^2.1.1"
    loose-envify "^1.0.0"
    object-assign "^4.1.0"
    promise "^7.1.1"
    setimmediate "^1.0.5"
    ua-parser-js "^0.7.9"

iconv-lite@~0.4.13:
  version "0.4.17"
  resolved "https://registry.yarnpkg.com/iconv-lite/-/iconv-lite-0.4.17.tgz#4fdaa3b38acbc2c031b045d0edcdfe1ecab18c8d"

is-stream@^1.0.1:
  version "1.1.0"
  resolved "https://registry.yarnpkg.com/is-stream/-/is-stream-1.1.0.tgz#12d4a3dd4e68e0b79ceb8dbc84173ae80d91ca44"

isomorphic-fetch@^2.1.1:
  version "2.2.1"
  resolved "https://registry.yarnpkg.com/isomorphic-fetch/-/isomorphic-fetch-2.2.1.tgz#611ae1acf14f5e81f729507472819fe9733558a9"
  dependencies:
    node-fetch "^1.0.1"
    whatwg-fetch ">=0.10.0"

j2c-plugin-prefix-browser@^1.0.6:
  version "1.0.6"
  resolved "https://registry.yarnpkg.com/j2c-plugin-prefix-browser/-/j2c-plugin-prefix-browser-1.0.6.tgz#bf36395d1dc3d95f3e3dacea9dca9829baa95fd1"

j2c@^1.0.0-4:
  version "1.0.0-4"
  resolved "https://registry.yarnpkg.com/j2c/-/j2c-1.0.0-4.tgz#a1c85b2b8bc4f71cb8f8bb53e8b396fa3fd899f0"

js-tokens@^3.0.0:
  version "3.0.1"
  resolved "https://registry.yarnpkg.com/js-tokens/-/js-tokens-3.0.1.tgz#08e9f132484a2c45a30907e9dc4d5567b7f114d7"

loose-envify@^1.0.0, loose-envify@^1.1.0, loose-envify@^1.3.1:
  version "1.3.1"
  resolved "https://registry.yarnpkg.com/loose-envify/-/loose-envify-1.3.1.tgz#d1a8ad33fa9ce0e713d65fdd0ac8b748d478c848"
  dependencies:
    js-tokens "^3.0.0"

node-fetch@^1.0.1:
  version "1.7.0"
  resolved "https://registry.yarnpkg.com/node-fetch/-/node-fetch-1.7.0.tgz#3ff6c56544f9b7fb00682338bb55ee6f54a8a0ef"
  dependencies:
    encoding "^0.1.11"
    is-stream "^1.0.1"

object-assign@^4.1.0, object-assign@^4.1.1:
  version "4.1.1"
  resolved "https://registry.yarnpkg.com/object-assign/-/object-assign-4.1.1.tgz#2109adc7965887cfc05cbbd442cac8bfbb360863"

polythene-core-css@^1.0.0-rc.4:
  version "1.0.0-rc.4"
  resolved "https://registry.yarnpkg.com/polythene-core-css/-/polythene-core-css-1.0.0-rc.4.tgz#4bd2183bcbce096e8d0dc22710cefe61b33cc28c"
  dependencies:
    j2c "^1.0.0-4"
    j2c-plugin-prefix-browser "^1.0.6"
    polythene-core "^1.0.0-rc.4"

polythene-core-shadow@^1.0.0-rc.5:
  version "1.0.0-rc.5"
  resolved "https://registry.yarnpkg.com/polythene-core-shadow/-/polythene-core-shadow-1.0.0-rc.5.tgz#5c950ce65c1c365116dc61459b1015f993afa6af"
  dependencies:
    polythene-core "^1.0.0-rc.4"
    polythene-core-css "^1.0.0-rc.4"
    polythene-theme "^1.0.0-rc.5"

polythene-core-spinner@^1.0.0-rc.5:
  version "1.0.0-rc.5"
  resolved "https://registry.yarnpkg.com/polythene-core-spinner/-/polythene-core-spinner-1.0.0-rc.5.tgz#53f3528661a466dedd8e65a0d724d21a50eea9b0"
  dependencies:
    polythene-core "^1.0.0-rc.4"
    polythene-core-css "^1.0.0-rc.4"
    polythene-theme "^1.0.0-rc.5"
    polythene-utilities "^1.0.0-rc.5"

polythene-core@^1.0.0-rc.4:
  version "1.0.0-rc.4"
  resolved "https://registry.yarnpkg.com/polythene-core/-/polythene-core-1.0.0-rc.4.tgz#3942453d959785671024909895a6677eab5b3b8b"

polythene-react-base@^1.0.0-rc.4:
  version "1.0.0-rc.4"
  resolved "https://registry.yarnpkg.com/polythene-react-base/-/polythene-react-base-1.0.0-rc.4.tgz#355d66d13386651e6a5d7600c41126d835e751fb"
  dependencies:
    polythene-core "^1.0.0-rc.4"

polythene-react-shadow@^1.0.0-rc.5:
  version "1.0.0-rc.5"
  resolved "https://registry.yarnpkg.com/polythene-react-shadow/-/polythene-react-shadow-1.0.0-rc.5.tgz#1bb568875d4518a497a81e8eb75fbdd3f8d21eb9"
  dependencies:
    polythene-core-shadow "^1.0.0-rc.5"
    polythene-react-base "^1.0.0-rc.4"

polythene-react-spinner@^1.0.0-rc.5:
  version "1.0.0-rc.5"
  resolved "https://registry.yarnpkg.com/polythene-react-spinner/-/polythene-react-spinner-1.0.0-rc.5.tgz#58b0dad615247586b7902da1e2a6fadbeec23be1"
  dependencies:
    polythene-core "^1.0.0-rc.4"
    polythene-core-spinner "^1.0.0-rc.5"
    polythene-react-base "^1.0.0-rc.4"
    polythene-react-shadow "^1.0.0-rc.5"

polythene-style@^1.0.0-rc.5:
  version "1.0.0-rc.5"
  resolved "https://registry.yarnpkg.com/polythene-style/-/polythene-style-1.0.0-rc.5.tgz#1c656c234c1384e921b2eef202c0cd48f4399537"
  dependencies:
    polythene-core "^1.0.0-rc.4"
    polythene-core-css "^1.0.0-rc.4"
    polythene-utilities "^1.0.0-rc.5"

polythene-theme@^1.0.0-rc.5:
  version "1.0.0-rc.5"
  resolved "https://registry.yarnpkg.com/polythene-theme/-/polythene-theme-1.0.0-rc.5.tgz#b8a891b2185458629999c2ac875624d634ca6c5a"
  dependencies:
    polythene-style "^1.0.0-rc.5"

polythene-utilities@^1.0.0-rc.5:
  version "1.0.0-rc.5"
  resolved "https://registry.yarnpkg.com/polythene-utilities/-/polythene-utilities-1.0.0-rc.5.tgz#5ddefdd3199c9600bbad739067cc31035797504b"
  dependencies:
    polythene-core "^1.0.0-rc.4"
    polythene-core-css "^1.0.0-rc.4"
    webfontloader "^1.6.27"

promise@^7.1.1:
  version "7.1.1"
  resolved "https://registry.yarnpkg.com/promise/-/promise-7.1.1.tgz#489654c692616b8aa55b0724fa809bb7db49c5bf"
  dependencies:
    asap "~2.0.3"

prop-types@^15.5.10:
  version "15.5.10"
  resolved "https://registry.yarnpkg.com/prop-types/-/prop-types-15.5.10.tgz#2797dfc3126182e3a95e3dfbb2e893ddd7456154"
  dependencies:
    fbjs "^0.8.9"
    loose-envify "^1.3.1"

react-hyperscript@^3.0.0:
  version "3.0.0"
  resolved "https://registry.yarnpkg.com/react-hyperscript/-/react-hyperscript-3.0.0.tgz#3c16010b33175de6bc01fd1ebad0a16a9a6dc9ab"

react@^15.6.1:
  version "15.6.1"
  resolved "https://registry.yarnpkg.com/react/-/react-15.6.1.tgz#baa8434ec6780bde997cdc380b79cd33b96393df"
  dependencies:
    create-react-class "^15.6.0"
    fbjs "^0.8.9"
    loose-envify "^1.1.0"
    object-assign "^4.1.0"
    prop-types "^15.5.10"

setimmediate@^1.0.5:
  version "1.0.5"
  resolved "https://registry.yarnpkg.com/setimmediate/-/setimmediate-1.0.5.tgz#290cbb232e306942d7d7ea9b83732ab7856f8285"

ua-parser-js@^0.7.9:
  version "0.7.12"
  resolved "https://registry.yarnpkg.com/ua-parser-js/-/ua-parser-js-0.7.12.tgz#04c81a99bdd5dc52263ea29d24c6bf8d4818a4bb"

webfontloader@^1.6.27:
  version "1.6.28"
  resolved "https://registry.yarnpkg.com/webfontloader/-/webfontloader-1.6.28.tgz#db786129253cb6e8eae54c2fb05f870af6675bae"

whatwg-fetch@>=0.10.0:
  version "2.0.3"
  resolved "https://registry.yarnpkg.com/whatwg-fetch/-/whatwg-fetch-2.0.3.tgz#9c84ec2dcf68187ff00bc64e1274b442176e1c84"
