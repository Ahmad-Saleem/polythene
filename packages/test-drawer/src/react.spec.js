import { runSnapshots } from "../../tests/scripts/react-snapshots";
import { renderer, keys, Drawer } from "polythene-react";
import specTests from "./spec-tests.js";
import reactTests from "./tests-react.js";

runSnapshots({
  tests: specTests({ Drawer, renderer, keys }).concat(reactTests),
  renderer
});
