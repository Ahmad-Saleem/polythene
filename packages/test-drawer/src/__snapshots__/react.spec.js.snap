exports[`test No options 1`] = `
<DrawerToggle
  instance={[Function]} />
`;

exports[`test Option: class 1`] = `
<DrawerToggle
  className="class-x"
  instance={[Function]} />
`;

exports[`test Option: className 1`] = `
<DrawerToggle
  className="className-x"
  instance={[Function]} />
`;

exports[`test Option: id 1`] = `
<DrawerToggle
  id="id-x"
  instance={[Function]} />
`;

exports[`test Permanent, floating (no shadow) 1`] = `
<Drawer
  content={
    <List
      compact={true}
      hoverable={true}
      tiles={
        Array [
          <ListTile
            className="tests-drawer-navigation-list"
            events={
                Object {
                    "onClick": [Function],
                  }
            }
            front={
                <Icon
                    svg={
                        Object {
                            "content": <div
                              dangerouslySetInnerHTML={
                                    Object {
                                          "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H4.99c-1.11 0-1.98.89-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.11-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10z\"/></svg>",
                                        }
                              } />,
                          }
                    } />
            }
            hoverable={true}
            navigation={true}
            title="Inbox" />,
          <ListTile
            className="tests-drawer-navigation-list"
            events={
                Object {
                    "onClick": [Function],
                  }
            }
            front={
                <Icon
                    svg={
                        Object {
                            "content": <div
                              dangerouslySetInnerHTML={
                                    Object {
                                          "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\"/></svg>",
                                        }
                              } />,
                          }
                    } />
            }
            hoverable={true}
            navigation={true}
            title="Starred" />,
          <ListTile
            className="tests-drawer-navigation-list"
            events={
                Object {
                    "onClick": [Function],
                  }
            }
            front={
                <Icon
                    svg={
                        Object {
                            "content": <div
                              dangerouslySetInnerHTML={
                                    Object {
                                          "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M2.01 21L23 12 2.01 3 2 10l15 2-15 2z\"/></svg>",
                                        }
                              } />,
                          }
                    } />
            }
            hoverable={true}
            navigation={true}
            title="Sent mail" />,
          <ListTile
            className="tests-drawer-navigation-list"
            events={
                Object {
                    "onClick": [Function],
                  }
            }
            front={
                <Icon
                    svg={
                        Object {
                            "content": <div
                              dangerouslySetInnerHTML={
                                    Object {
                                          "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M21.99 8c0-.72-.37-1.35-.94-1.7L12 1 2.95 6.3C2.38 6.65 2 7.28 2 8v10c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2l-.01-10zM12 13L3.74 7.84 12 3l8.26 4.84L12 13z\"/></svg>",
                                        }
                              } />,
                          }
                    } />
            }
            hoverable={true}
            navigation={true}
            title="Drafts" />,
        ]
      } />
  }
  permanent={true}
  z={0} />
`;

exports[`test Permanent, floating (shadow depth 1) 1`] = `
<Drawer
  content={
    <List
      compact={true}
      hoverable={true}
      tiles={
        Array [
          <ListTile
            className="tests-drawer-navigation-list"
            events={
                Object {
                    "onClick": [Function],
                  }
            }
            front={
                <Icon
                    svg={
                        Object {
                            "content": <div
                              dangerouslySetInnerHTML={
                                    Object {
                                          "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H4.99c-1.11 0-1.98.89-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.11-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10z\"/></svg>",
                                        }
                              } />,
                          }
                    } />
            }
            hoverable={true}
            navigation={true}
            title="Inbox" />,
          <ListTile
            className="tests-drawer-navigation-list"
            events={
                Object {
                    "onClick": [Function],
                  }
            }
            front={
                <Icon
                    svg={
                        Object {
                            "content": <div
                              dangerouslySetInnerHTML={
                                    Object {
                                          "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\"/></svg>",
                                        }
                              } />,
                          }
                    } />
            }
            hoverable={true}
            navigation={true}
            title="Starred" />,
          <ListTile
            className="tests-drawer-navigation-list"
            events={
                Object {
                    "onClick": [Function],
                  }
            }
            front={
                <Icon
                    svg={
                        Object {
                            "content": <div
                              dangerouslySetInnerHTML={
                                    Object {
                                          "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M2.01 21L23 12 2.01 3 2 10l15 2-15 2z\"/></svg>",
                                        }
                              } />,
                          }
                    } />
            }
            hoverable={true}
            navigation={true}
            title="Sent mail" />,
          <ListTile
            className="tests-drawer-navigation-list"
            events={
                Object {
                    "onClick": [Function],
                  }
            }
            front={
                <Icon
                    svg={
                        Object {
                            "content": <div
                              dangerouslySetInnerHTML={
                                    Object {
                                          "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M21.99 8c0-.72-.37-1.35-.94-1.7L12 1 2.95 6.3C2.38 6.65 2 7.28 2 8v10c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2l-.01-10zM12 13L3.74 7.84 12 3l8.26 4.84L12 13z\"/></svg>",
                                        }
                              } />,
                          }
                    } />
            }
            hoverable={true}
            navigation={true}
            title="Drafts" />,
        ]
      } />
  }
  permanent={true}
  z={1} />
`;
